devicetypes/smartthings/zigbee-battery-accessory-dimmer.src/zigbee-battery-accessory-dimmer.groovy

```diff
@@ -0,0 +1,151 @@
+			}
+			tileAttribute ("device.level", key: "SLIDER_CONTROL") {
+				attributeState "level", action:"switch level.setLevel"
+			}
+			tileAttribute ("device.battery", key: "SECONDARY_CONTROL") {
+				attributeState "battery", label: 'battery ${currentValue}%', unit: "%"
+			}
+		}
+		main "switch"
+		details(["switch"])
+	}
+}
+
+def getDOUBLE_STEP() {10}
+def getSTEP() {5}
+
+def parse(String description) {
+
+	def descMap = zigbee.parseDescriptionAsMap(description)
+	if(isSengledSwitch()) {
+		handleSengledSwitchEvents(descMap)
+	}
+	handleBatteryEvents(descMap)
+}
+
+
+def handleSengledSwitchEvents(descMap) {
+	if (descMap && descMap.clusterInt == 0xFC10) {
+		def currentLevel = device.currentValue("level") as Integer ?: 0
+		def value = 0
+
+		switch(descMap.data[0]) {
+			case '01':
+				//short press of 'ON' button
+				sendEvent(name: "switch", value: "on")
+				break
+			case '02':
+				// move up
+				if (descMap.data[2] == '02') {
+					//long press of 'BRIGHTEN' button
+					value = Math.min(currentLevel + DOUBLE_STEP, 100)
+				} else if (descMap.data[2] == '01') {
+					//short press of 'BRIGHTEN' button
+					value = Math.min(currentLevel + STEP, 100)
+				}
+				sendEvent(name: "switch", value: "on")
+				sendEvent(name: "level", value: value)
+				break
+			case '03':
+				//move down
+				if (descMap.data[2] == '02') {
+					//long press of 'DIM' button
+					value = Math.max(currentLevel - DOUBLE_STEP, 0)
+				} else if (descMap.data[2] == '01') {
+					//short press of 'DIM' button
+					value = Math.max(currentLevel - STEP, 0)
+				}
+
+				if (value == 0) {
+					sendEvent(name: "switch", value: "off")
+				}
+				sendEvent(name: "level", value: value)
+				break
+			case '04':
+				//short press of 'OFF' button
+				sendEvent(name: "switch", value: "off")
+				break
+			case '06':
+				//long press of 'ON' button
+				 sendEvent(name: "switch", value: "off")
+				 break
+			case '08':
+				//long press of 'OFF' button
+				sendEvent(name: "switch", value: "off")
+				break
+			default:
+				break
+		}
+	}
+}
+
+def handleBatteryEvents(descMap) {
+	if (descMap && descMap.clusterInt == zigbee.POWER_CONFIGURATION_CLUSTER && descMap?.value) {
+		def batteryValue = zigbee.convertHexToInt(descMap.value) / 2
+		sendEvent(name: "battery", value: batteryValue)
+	}
+}
+
+def off() {
+	sendEvent(name: "switch", value: "off", isStateChange: true)
+}
+
+def on() {
+	sendEvent(name: "switch", value: "on", isStateChange: true)
+}
+
+def setLevel(value, rate = null) {
+	if (value == 0) {
+		sendEvent(name: "switch", value: "off")
+	} else {
+		sendEvent(name: "switch", value: "on")
+		sendEvent(name: "level", value: value)
+	}
+}
+
+def installed() {
+	sendEvent(name: "switch", value: "on", displayed: false)
+	sendEvent(name: "level", value: 100, displayed: false)
+}
+
+def configure() {
+	zigbee.readAttribute(0x0001, 0x0021) + zigbee.configureReporting(0x0001, 0x0021, DataType.UINT8, 30, 6 * 60 * 60, null)
```

Comment:
Use `zigbee.POWER_CONFIGURATION_CLUSTER` instead of 0x0001

Reply:
ok